Automattic/jetpack
# Nav Unification: Store preferred view on page switch

Fixes https://github.com/Automattic/wp-calypso/issues/53709

#### Changes proposed in this Pull Request:

Stores the preferred view when a customer uses the WP.com quick switcher for switching from Calypso to WP Admin or from WP Admin to Calypso.

#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Set up the testing environment:
  - Simple sites: Apply D63506-code to your WP.com sandbox, and sandbox both the API and a Simple site.
  - Atomic sites: Install Jetpack Beta and activate the `update/nav-unification-store-preferred-view` branch.
- Go to https://wordpress.com.
- Switch to your Simple or Atomic site.
- Go to any of the following screens:
  - Posts > All posts
  - Posts > Categories
  - Posts > Tags
  - Media
  - Pages > All pages
  - Testimonials > All testimonials
  - Portfolio > All projects
  - Comments
  - Appearance > Themes
  - Users > All users
  - Tools > Import
  - Tools > Export
  - Settings > General
  - Settings > Writing
  - Settings > Discussion
- Click on the "Screen Options" tab and then select "Classic view".
- Make sure you're redirected to the equivalent WP Admin screen.
- Make sure that the menu/submenu item of that screen now always links to the WP Admin screen from both Calypso and WP Admin.
- On the WP Admin screen, click on the "Screen Options" tab and then select "Use WordPress.com view".
- Make sure you're redirected to the equivalent Calypso screen.
- Make sure that the menu/submenu item of that screen now always links to the Calypso screen from both Calypso and WP Admin.
